crates/crypto/src/key_manager/local_key_manager.rs

```diff
@@ -0,0 +1,127 @@
+use crate::key::{KeyType, PrivateKey, PublicKey};
+use crate::key_manager::key_store::{InMemoryKeyStore, KeyStore};
+use crate::key_manager::{KeyManager, KeyManagerError};
+use ssi_jwk::JWK;
+use std::sync::Arc;
+
+/// Implementation of the [`KeyManager`] trait with key generation local to the device/platform it
+/// is being run. Key storage is provided by a [`KeyStore`] trait implementation, allowing the keys
+/// to be stored wherever is most appropriate for the application.
+pub struct LocalKeyManager {
+    key_store: Arc<dyn KeyStore>,
+}
+
+impl LocalKeyManager {
+    /// Constructs a new `LocalKeyManager` that stores keys in the provided `KeyStore`.
+    pub fn new(key_store: Arc<dyn KeyStore>) -> Self {
+        Self { key_store }
```

Comment:
Is this a place where you should be using `::clone` on the passed in `key_store`? If not, mind elaborating why?

Reply:
No, not exactly. The contract here says "I need an `Arc<dyn KeyStore>`." It's the caller's responsibility to provide that. Whether that's done by a `::clone()` or `::new()` is an implementation detail that the caller must handle. If they do not, it's a compilation error.